src/pages/SearchPage/SearchPage.js

```diff
@@ -26,39 +27,59 @@ function parseRedditData(input) {
     arrInfo[day][hour].push({
       author: result.author,
       title: result.title,
-      created_utc: result.created_utc,
+      created: date,
       score: result.score,
       num_comments: result.num_comments,
+      full_link: result.full_link,
     });
   });
   return arrInfo;
 }
 
 function SearchPage() {
-  const [isLoading, setIsLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
   const [error, setError] = useState('');
   const [searchResults, setSearchResults] = useState([]);
   const [heatmapInfo, setHeatmapInfo] = useState([]);
-  const { subreddit } = useParams();
+  const history = useHistory();
 
-  // Search for new subreddit when parameters have changed
+  // Load data
+  //    Will load data if there:
+  //      1. isn't a current load in place,
+  //      2. is a change to history detected,
+  //      3. the useParams() property changes
   useEffect(() => {
-    setIsLoading(true);
-    setError('');
-    handleSearch(subreddit)
-      .then((data) => {
-        if (data.data.length !== 0) {
-          setSearchResults(data);
-        } else {
-          setError('0 results returned.');
-        }
-      })
-      .catch(() => {
-        setError('Error loading data.');
-      })
-      .finally(() => setIsLoading(false));
-  }, [subreddit]);
+    function loadData() {
+      const searchPath = history.location.pathname.split('/')[2] || DEFAULT_SUBREDDIT;
+      setIsLoading(true);
+      setError('');
 
+      handleSearch(searchPath)
+        .then((data) => {
+          if (data.data.length !== 0) {
+            setSearchResults(data);
+          } else {
+            setError('0 results returned.');
+          }
+        })
+        .catch(() => {
+          setError('Error loading data.');
+        })
+        .finally(() => setIsLoading(false));
+    }
+
+    if (!isLoading) {
```

Comment:
This doesn't have an effect btw. It's a stale closure. `isLoading` will always have the value of the first render (meaning `false`) since it's not part of the dependencies. As I said, you can simply remove it since I don't think you need to prevent data loading except from diasbling the button.

Or is there more to it? In any case, the current implementation will always call loadData no matter how the `isLoading` state looks like.